File: Maths/sommeChiffres.js
/**
 * @description Fonction qui calcule la somme des chiffres d'un nombre
 * @param {number} n - Nombre à calculer
 * @returns {number} - Somme des chiffres du nombre
 */
const sommeChiffres = (n) => {
    let somme = 0;
    while (n > 0) {
        somme += n % 10;
        n = ~~(n / 10);
    }
    return somme;
    }

// Test
console.log(sommeChiffres(1010101)); // 4

File: Python/Maths/somme_chiffres.py
def somme_chiffres(n: int) -> int:
    """
    Description:
        Fonction qui retourne la somme des chiffres d'un nombre.
    
    Paramètres:
        n: {int} -- Le nombre à traiter.
    
    Retourne:
        {int} -- La somme des chiffres du nombre.
    
    Exemple:
        >>> somme_chiffres(1010101)
        4
    """
    somme = 0
    while n > 0:
        somme += n % 10
        n //= 10
    return somme